R/FilterStates.R

```diff
@@ -165,7 +165,12 @@ FilterStates <- R6::R6Class( # nolint
       )
       if (length(filter_items) > 0L) {
         filter_function <- private$fun
-        data_name <- str2lang(private$dataname_prefixed)
+        data_name <- tryCatch(
```

Comment:
I think it should be done where `dataname_prefixed` is created. Here it is little too late as dataname_prefixed could be `multi assay experiment[["experiment 1"]]`. Please search for `private$dataname_prefixed <-`. Also, I think it is better to not handle error but

```
private$dataname_prefixed <- if (!identical(datanames, make.names(dataname)) {
  sprintf("`%s`", dataname)
} else {
  dataname
}
```

Reply:
Yup, that's right and it still needs some more extra conversions.

I'm playing with ``format(as.name("`yada`"))`` instead of that if clause

My tests today with MAE are also failing, initially because of this error, but also downstream in `FilterState$private$get_varname_prefixed`.

<details>

<summary>App</summary>

```r
pkgload::load_all("teal.data")
pkgload::load_all("teal.slice")
pkgload::load_all("teal")


options(
  teal.log_level = "TRACE",
  teal.log_layout = paste(
    "{crayon::bold(logger::colorize_by_log_level(level, levelr))}",
    "from {namespace}",
    "[{crayon::italic(format(time, \"%Y-%m-%d %H:%M:%S\"))}]",
    "{logger::grayscale_by_log_level(msg, levelr)}"
  ),
  teal.show_js_log = TRUE,
  # teal.bs_theme = bslib::bs_theme(version = 5),
  shiny.bookmarkStore = "server"
)

app <- teal::init(
  data = teal_data() |> within({
    `iris raw` <- cbind(iris, data.frame(date = Sys.Date() + sample(seq(100), nrow(iris), replace = TRUE)))
    `%iris%` <- iris
    env_temp <- new.env()
    data("miniACC", package = "MultiAssayExperiment", envir = env_temp)
    `a MAE` <- env_temp$miniACC
    miniACC_copy <- env_temp$miniACC
    rm(env_temp)
  }), 
  # filter = teal_slices(
  #   teal_slice("iris raw", "Species", selected = "setosa")
  # ),
  modules = modules(
    example_module("exotic"),
    example_module("exotic-specified", datanames = "iris raw")
  )
)

runApp(app)
```

</details>

Reply:
Works now for MAE and Matrix.

Flagged a new bug here that was already present in `teal.slice`

https://github.com/insightsengineering/teal.slice/issues/625